k2/csrc/context.h

```diff
@@ -329,6 +321,10 @@ ContextPtr GetCpuContext();
 // from PyTorch
 ContextPtr GetCudaContext(int32_t gpu_id = -1);
 
+// Returns a (CPU) context that will allocate pinned memory.  (This is CPU
+// memory that's pinned for faster GPU memory transfers).  May or may not
+// return the same value as ::k2::GetCpuContext()... this is so, for instance,
+// if you have a GPU PyTorch context you can get a CPU PyTorch context.
 // Returns a (CPU) context that will allocate pinned memory.
 ContextPtr GetPinnedContext();
```

Comment:
I'm thinking we could rename this to `GetHostContext()`, and say: if `this` is a CPU context it will just return a pointer to `this`, but if `this` is a GPU context it will return a pinned context from which we can do asynchronous transfers.

Reply:
`GetPinnedContext ` is not a member method so there is no `this` pointer. The returned context's GetDeviceType()
always returns `kCPU` and it is compatible with contexts returned from `GetCpuContext()`.

Do we still need to rename it to `GetHostContext()`?  I think there are confusions between `GetCpuContext()`
and `GetHostContext()`. 

From #451 
> In addition to the contexts returned by GetCpuContext() and GetCudaContext(), we need a 3rd type of context which would return pinned (page-locked) memory. Call this function GetPinnedContext().
(We need to remove the member-function of the same name from class Context).

Reply:
Mm.  If it's non-member, GetPinnedContext makes the most sense as a function name.
We'll need a not-too-wordy mechanism to do the logic of:
If the input is CUDA, give a pinned context, else give a CPU context.
which will commonly be needed.

Reply:
Added `GetContextForTransfer`